search-filter.lua

```diff
@@ -0,0 +1,108 @@
+-- Required libraries
+local gui = require('gui') -- Importing the 'gui' library
+local widgets = require('gui.widgets') -- Importing the 'widgets' library
+
+-- Define SearchEngine class
+SearchEngine = defclass(SearchEngine, widgets.Window) -- Defining a new class 'SearchEngine' that inherits from 'widgets.Window'
+SearchEngine.ATTRS = {
+    frame_title='Search Engine for Fort-Controlled Units', -- Title of the frame
+    frame={w=50, h=45}, -- Dimensions of the frame
+    resizable=true, -- Frame can be resized
+    resize_min={w=43, h=20}, -- Minimum dimensions when resizing
+}
+
+function SearchEngine:init() -- Initialization function for the SearchEngine class
+    self:addviews{
+        widgets.EditField{
+            view_id='edit',
+            frame={t=1, l=1}, -- Position of the EditField view
+            text='', -- Initial text in the EditField view
+            on_change=self:callback('updateList'), -- Callback function when text in EditField changes
+        },
+        widgets.List{
+            view_id='list',
+            frame={t=3, b=0},
+            choices=self:getFortControlledUnits(), -- Choices in the List view are obtained from getFortControlledUnits function
+            on_select=self:callback('onSelect'), -- Callback function when a unit is selected from the list
+        }
+    }
+end
+
+function SearchEngine:getFortControlledUnits()
+    local fortControlledUnits = {}
+    for _, unit in ipairs(df.global.world.units.active) do
+        local unitName = dfhack.TranslateName(dfhack.units.getVisibleName(unit))
+        if unitName == "" then
+            -- Use the race name if the unit's name field is empty
+            unitName = dfhack.units.getRaceName(unit)
+        end
```

Comment:
`dfhack.units.getReadableName(unit)` would be a better fit here

Reply:
I added excessive comments because I'm not a programmer. These comments in the code are most likely wrong, and I'm open to having them all removed.

Regarding the program's functionality, it successfully realized my initial vision, which makes me satisfied. Additionally, I believe it's crucial to maintain the graphical user interface for the search filter program, as it offers an accessibility feature that I would greatly miss.

Reply:
Also I'm not interested in any recognition/credit so if someone is interested in taking all the code and calling it their own. I'm 100% happy with that. I just want others to enjoy the ability to ctrl+f aka use a search engine.

Reply:
Also feel free to contact me on discord: unboundlopez